polyphemus/lib/etls/metis_files_linker_base.rb

```diff
@@ -107,13 +109,13 @@ def files_payload(project_name:, model_name:, files:, attribute_regex:, record:,
         next if payloads_for_attr.empty?
 
         if is_file_collection?(project_name, model_name, attribute_name)
-          if record.nil? || (existing_files = record[attribute_name]).nil?
+          if record.nil? || (existing_files = record[attribute_name.to_s]).nil?
             payload[attribute_name] = payloads_for_attr
           else
-            payload_by_paths = payloads_for_attr.map { |f| f[:path] }
-            payload[attribute_name] = payloads_for_attr + existing_files.select do |file|
-              !payload_by_paths.include?(file[:path])
-            end
+            payload_by_filename = payloads_for_attr.map { |f| f["original_filename"] }
```

Comment:
Changed to use the `original_filename` attribute to do comparison by. Unfortunately really nothing in a `file_collection` attribute needs to be unique except the index location within the Array, so I don't think there's really a good way to determine uniqueness. We can't rely on `path` because for the existing files, the Magma "path" will be renamed according to the attribute (i.e. a file called "original.txt" on Metis will get renamed to "rna_seq-RECORD_NAME-raw_fastq_files-0.txt" on Magma in the `path` value). Only the `original_filename` attribute will remain "like" the Metis file name and be comparable...